keel-core/src/main/kotlin/com/netflix/spinnaker/keel/lifecycle/LifecycleEventHandler.kt

```diff
@@ -1,18 +1,27 @@
 package com.netflix.spinnaker.keel.lifecycle
 
+import org.springframework.context.ApplicationEventPublisher
 import org.springframework.context.event.EventListener
 import org.springframework.stereotype.Component
 
 /**
- * Receives lifecycle events and stores them
+ * Receives lifecycle events and stores them.
  */
 @Component
 class LifecycleEventHandler(
-  val repository: LifecycleEventRepository
+  val repository: LifecycleEventRepository,
+  val publisher: ApplicationEventPublisher
 ) {
 
+  /**
+   * Saves all lifecycle events.
+   * If they should be monitored, publish a [StartMonitoringEvent]
+   */
   @EventListener(LifecycleEvent::class)
   fun handleEvent(event: LifecycleEvent) {
-    repository.saveEvent(event)
+    val eventId = repository.saveEvent(event)
+    if (event.startMonitoring) {
+      publisher.publishEvent(StartMonitoringEvent(eventId, event))
+    }
   }
```

Comment:
now we publish a `StartMonitoringEvent` to kick off monitoring instead of relying on another handler to also be listening to `LifecyceEvent`s